util/stopper.go

```diff
@@ -51,17 +50,20 @@ type Stopper struct {
 	stopped  chan struct{}  // Closed when stopped completely
 	stop     sync.WaitGroup // Incremented for outstanding workers
 	mu       sync.Mutex     // Protects the fields below
```

Comment:
Hmm, I got a nil pointer dereference error. It seems we still need to call `sync.NewCond`. Also, it might be a bit confusing if `drain.L` is used to protect `closers`, etc... I'll keep the code as-is. (We could embed `sync.Mutex`, but I wasn't sure if it's better.)